src/Compilers/CSharp/Portable/Symbols/Source/SourceMethodSymbolWithAttributes.cs

```diff
@@ -0,0 +1,945 @@
+                    return data != null ? data.ObsoleteAttributeData : null;
+                }
+
+                if (syntaxReferenceOpt is null)
+                {
+                    // no references -> no attributes
+                    return null;
+                }
+
+                return ObsoleteAttributeData.Uninitialized;
+            }
+        }
+
+        internal sealed override ImmutableArray<string> GetAppliedConditionalSymbols()
+        {
+            CommonMethodEarlyWellKnownAttributeData data = this.GetEarlyDecodedWellKnownAttributeData();
+            return data != null ? data.ConditionalSymbols : ImmutableArray<string>.Empty;
+        }
+
+        internal sealed override void DecodeWellKnownAttribute(ref DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments)
+        {
+            Debug.Assert(!arguments.Attribute.HasErrors);
+            Debug.Assert(arguments.SymbolPart == AttributeLocation.None || arguments.SymbolPart == AttributeLocation.Return);
+
+            if (arguments.SymbolPart == AttributeLocation.None)
+            {
+                DecodeWellKnownAttributeAppliedToMethod(ref arguments);
+            }
+            else
+            {
+                DecodeWellKnownAttributeAppliedToReturnValue(ref arguments);
+            }
+        }
+
+        private void DecodeWellKnownAttributeAppliedToMethod(ref DecodeWellKnownAttributeArguments<AttributeSyntax, CSharpAttributeData, AttributeLocation> arguments)
+        {
+            Debug.Assert((object)arguments.AttributeSyntaxOpt != null);
+
+            var attribute = arguments.Attribute;
+            Debug.Assert(!attribute.HasErrors);
+
+            if (attribute.IsTargetAttribute(this, AttributeDescription.PreserveSigAttribute))
+            {
+                arguments.GetOrCreateData<MethodWellKnownAttributeData>().SetPreserveSignature(arguments.Index);
+            }
+            else if (attribute.IsTargetAttribute(this, AttributeDescription.MethodImplAttribute))
+            {
+                AttributeData.DecodeMethodImplAttribute<MethodWellKnownAttributeData, AttributeSyntax, CSharpAttributeData, AttributeLocation>(ref arguments, MessageProvider.Instance);
+            }
+            else if (attribute.IsTargetAttribute(this, AttributeDescription.DllImportAttribute))
+            {
+                DecodeDllImportAttribute(ref arguments);
+            }
+            else if (attribute.IsTargetAttribute(this, AttributeDescription.SpecialNameAttribute))
+            {
+                arguments.GetOrCreateData<MethodWellKnownAttributeData>().HasSpecialNameAttribute = true;
+            }
+            else if (attribute.IsTargetAttribute(this, AttributeDescription.ExcludeFromCodeCoverageAttribute))
+            {
+                arguments.GetOrCreateData<MethodWellKnownAttributeData>().HasExcludeFromCodeCoverageAttribute = true;
```

Comment:
>arguments.GetOrCreateData<MethodWellKnownAttributeData>().HasExcludeFromCodeCoverageAttribute = true; [](start = 16, length = 101)

Should we adjust implementation of DynamicAnalysisInjector.IsExcludedFromCodeCoverage?

Reply:
Adding a PROTOTYPE comment for it

---
In reply to: [354051939](https://github.com/dotnet/roslyn/pull/40011#discussion_r354051939) [](ancestors = 354051939)